src/lib.rs

```diff
@@ -2341,6 +2341,37 @@ impl Connection {
 
                     self.peer_transport_params = peer_params;
 
+                    // Update max_data of current sender streams with some data
+                    // already written. This can happen when there is a
+                    // Handshake packet loss but received actual request already,
+                    // so the stream (usually id 0) is already opened and have
+                    // some data to write, but actual
+                    // packetization is prevented because of
+                    // stream.send.max_tx_data = 0.
+                    // What we do here is update stream.send.max_data if
+                    // max_data=0 to its initial value used in
+                    // Stream::new(). In normal situation
+                    // (handshake is complete before getting
+                    // request) it will not do anything because
+                    // send.get_max_data() != 0
+                    for (id, stream) in self
+                        .streams
+                        .iter_mut()
+                        .filter(|(_, s)| s.send.get_max_data() == 0)
+                    {
+                        // we use same value of Stream::new()
+                        stream.send.update_max_data(
+                            self.peer_transport_params
+                                .initial_max_stream_data_bidi_local
+                                as usize,
+                        );
+                        trace!(
+                            "stream_id={} update stream.send.max_data={}",
```

Comment:
we probably want to log `&self.trace_id` too, so that server logs can disambiguate which connection is seeing this behaviour

Reply:
Updated